lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -85,13 +88,19 @@ def get_collection_versions(self, requirement):
         # type: (Requirement) -> Iterable[Tuple[str, GalaxyAPI]]
         """Get a set of unique versions for FQCN on Galaxy servers."""
         if requirement.is_concrete_artifact:
-            return {
-                (
-                    self._concrete_art_mgr.
-                    get_direct_collection_version(requirement),
-                    requirement.src,
-                ),
-            }
+            version = self._concrete_art_mgr.get_direct_collection_version(requirement)
+
+            err = f"Invalid version found for requirement '{requirement}': {version} ({type(version)}). "
+            err += "The version should be a semantic version string (or '*')."
+            if not isinstance(version, string_types):
+                raise AnsibleError(err)
+            elif version != "*":
+                try:
+                    SemanticVersion(version)
+                except ValueError as ex:
+                    raise AnsibleError(err) from ex
```

Comment:
Looking at this, I realize that it could be far more valuable to have a more specific error message for a "non-semantic version" vs "a non-string thing in YAML". For these, I'd also expect separate test cases.

Reply:
I wanted the error to describe a valid version, regardless of what was specifically wrong with it. What error message would you prefer for non-semantic versions?

Reply:
I'm thinking about the scenario of someone just quoting the float if they get an error about needing a string; then they'd get a different error.

Reply:
Well, why not show exactly that? Something along the lines of "A SemVer-compliant version expected but XXX found. See https://semver.org to learn how to compose it correctly."

Reply:
Ok, but do you want different error messages for both? And why?

Reply:
Nice! That seems clear.